actors/builtin/miner/expiration_queue.go

```diff
@@ -380,10 +403,13 @@ func (q ExpirationQueue) RemoveSectors(sectors []*SectorOnChainInfo, faults *abi
 			return false, false, err
 		}
 
-		// This loop could alternatively be done by constructing bitfields and intersecting them, but it's not
-		// clear that would be much faster (O(max(N, M)) vs O(N+M)).
-		// The length of sectors has a maximum of one partition size.
-		for _, sector := range sectors {
+		// NOTE: this iterates over the remaining _map_ to avoid
+		// revisiting already visited sectors. This is _ok_ because all
+		// the state updates we're doing here are idempotent.
+		//
+		// DO NOT update AMTs, HAMTs, etc. from within this loop. Only
+		// bitfields, big.Ints, etc.
+		for _, sector := range remaining { //nolint:nomaprange
```

Comment:
We could alternatively accumulate a `faultySectors` slice exactly where the `activeSectors` slice is accumulated, and then traverse safely without this warning.

Reply:
Welllll... I was trying to save a bit of time (avoid traversing sectors we've already found). But honestly, it's probably faster to create a slice and traverse it (because iterating over a slice is faster).